### PR TITLE
fix(portmapper): enforce timeouts for upnp

### LINKED ISSUE #2876: iroh does not work with relays without STUN on networks without UPnP
To reproduce, create `iroh.toml` file in the root of this repo checkout:
```
$ cat iroh.toml
[[relay_nodes]]
url = "https://iroh.testrun.org.:4443"
stun_only = false
stun_port = 3478
```

https://iroh.testrun.org.:4443 is running iroh-relay 0.25.0.

Then run:
```
$ cargo run -p iroh-cli -- --config ./iroh.toml doctor accept
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.27s
     Running `target/debug/iroh --config ./iroh.toml doctor accept`
Error: wait for relay connection

Caused by:
    deadline has elapsed
```

On a network with UPNP it works:
```
$ cargo run -p iroh-cli -- --config ./iroh.toml doctor accept
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.29s
     Running `target/debug/iroh --config ./iroh.toml doctor accept`
Connect to this node using one of the following commands:

	Using the relay url and direct connections:
iroh doctor connect ...
...
```

The problem is present since iroh 0.27.0. More precisely, commit https://github.com/n0-computer/iroh/commit/5e40fe138f9581a195d47c251992e3de8b1ec8c1, PR https://github.com/n0-computer/iroh/pull/2804

This PR pulled in commit https://github.com/dariusc93/rust-igd/commit/56934bb216149d1988c333c61e7c1d88a1868eb8 of rust-igd (`igd-next` crate) that was merged from PR https://github.com/dariusc93/rust-igd/pull/8

### PR DESCRIPTION
**Pull request by dignifiedquire**
Closes #2876

### CODE REVIEW

File: iroh.toml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[[relay_nodes]]
+url = "https://iroh.testrun.org.:4443"
```
Comment (dignifiedquire) [inline iroh.toml:2]: TODO: remove this file before merging

**Comment by matheus23** [comment]
What do you think of merging #2880 into this PR, then merging this? I think that's probably how we should do this.

**Comment by dignifiedquire** [comment]
> What do you think of merging #2880 into this PR, then merging this? I think that's probably how we should do this.

yes, please do

**Review by divagant-martian — APPROVED** [review]
LGTM!

**Pull request merged**